src/main.rs

```diff
@@ -165,8 +140,139 @@ fn main() -> Result<()> {
                 Err(err) => error!("{}", err),
             };
         }
+        ("history", Some(path)) => history_processing(path)?,
         (&_, _) => {}
     };
 
     Ok(())
 }
+
+fn history_processing(matches: &ArgMatches) -> Result<()> {
+    let histpath = PathBuf::from(
+        &matches
+            .value_of("history_parse")
+            .expect("Path to history file is not provided"),
+    );
+    let hist_file = File::open(histpath)?;
+    let reader = BufReader::new(hist_file);
+    let mut catches: Vec<Package> = reader
+        .lines()
+        .filter_map(|x| x.ok())
+        .map(|x| x.split_whitespace().join(" "))
+        .sorted()
```

Comment:
Can you explain what happens here? Why is the list being sorted twice?

Reply:
First time, I tokenize whole history, remove all whitespaces and join it back, than i sort  ans dedup` Vec<String>`. It's done in order not to envoke `catch` function several times on same data. 
Like `'vec!["yay -S\ \  lol", "yay    -S lol"] -> vec!["yay -S lol"]`
Second time, I sort and dedup `Vec<Packages>`